pyodide-build/pyodide_build/buildpkg.py

```diff
@@ -106,129 +134,240 @@ def check_checksum(path: Path, pkg: Dict[str, Any]):
         raise ValueError("Invalid {} checksum".format(checksum_algorithm))
 
 
+def trim_archive_extension(tarballname):
+    for extension in [
+        ".tar.gz",
+        ".tgz",
+        ".tar",
+        ".tar.bz2",
+        ".tbz2",
+        ".tar.xz",
+        ".txz",
+        ".zip",
+    ]:
+        if tarballname.endswith(extension):
+            return tarballname[: -len(extension)]
+    return tarballname
+
+
 def download_and_extract(
-    buildpath: Path, packagedir: Path, pkg: Dict[str, Any], args
+    buildpath: Path, srcpath: Path, src_metadata: Dict[str, Any]
 ) -> Path:
-    srcpath = buildpath / packagedir
+    """
+    Download the source from specified in the meta data, then checksum it, then
+    extract the archive into srcpath.
+
+    Arguments
+    ---------
```

Comment:
```suggestion
    Parameters
    ----------
```
(there is a spec for section names in numpy docstyle)